framework/src/modules/chain/dpos/delegates_list.js

```diff
@@ -74,18 +74,17 @@ class DelegatesList extends EventEmitter {
 		);
 
 		if (!delegatePublicKeys.length) {
+			const { ignoreDelegateListCacheForRounds = [] } = this.exceptions;
 			delegatePublicKeys = await this.getDelegatePublicKeysSortedByVote();
-			await this.storage.entities.RoundDelegates.create({
-				round,
-				delegatePublicKeys,
-			});
-		}
-
-		const { ignoreDelegateListCacheForRounds = [] } = this.exceptions;
 
-		if (!ignoreDelegateListCacheForRounds.includes(round)) {
-			// If the round is not an exception, cache the round.
-			this.delegateListCache[round] = delegatePublicKeys;
+			// If the round is not an exception, create entry in the database and cache the list.
+			if (!ignoreDelegateListCacheForRounds.includes(round)) {
```

Comment:
Nice workaround! 👍

Reply:
Just cross my mind that we probably don't need it anymore as the exception is being handled in `verifyBlockForger`.

@shuse2 What you reckon?

Reply:
yes we shouldnt do this on the generate part